pkg/controller/combo_controller.go

```diff
@@ -2,36 +2,134 @@ package controller
 	log := c.log.WithValues("request", req)
-	log.Info("reconciling combination")
 
-	in := &v1alpha1.Combination{}
-	if err := c.Get(ctx, req.NamespacedName, in); err != nil {
+	log.Info("new combination event inbound")
+
+	// Attempt to retrieve the requested combination CR
+	combination := &v1alpha1.Combination{}
+	err := c.Get(ctx, req.NamespacedName, combination)
+	if err != nil {
 		return reconcile.Result{}, client.IgnoreNotFound(err)
 	}
 
+	// If combination is being deleted, remove from queue
+	if !combination.ObjectMeta.DeletionTimestamp.IsZero() {
+		log.Info("combination is being deleted, ignoring event")
+		return reconcile.Result{}, nil
+	}
+
+	log.Info(fmt.Sprintf("combination %s successfully found in reconciler", combination.Name))
+
+	// Attempt to retrieve the template referenced in the combination CR
+	template := &v1alpha1.Template{}
+	templateQuery := types.NamespacedName{Name: combination.Spec.Template}
+	if err := c.Get(ctx, templateQuery, template); err != nil {
+		err = fmt.Errorf("failed to retrieve %v template: %w", combination.Spec.Template, err)
+		return c.manageFailure(ctx, combination, comboConditions.TemplateNotFoundCondition, err)
+	}
+
+	// Build combination stream to be utilized in template builder
+	comboStream := combinationPkg.NewStream(
+		combinationPkg.WithArgs(formatArguments(combination.Spec.Arguments)),
+		combinationPkg.WithSolveAhead(),
+	)
+
+	// Create a new template builder
+	builder, err := templatePkg.NewBuilder(strings.NewReader(template.Spec.Body), comboStream)
+	if err != nil {
+		err = fmt.Errorf("failed to construct a builder out of %s template body: %w", template.Name, err)
+		return c.manageFailure(ctx, combination, comboConditions.TemplateBodyInvalid, err)
+	}
+
+	// Build the manifest combinations
+	generatedManifests, err := builder.Build(ctx)
+	if err != nil {
+		err = fmt.Errorf("failed to generate manifest %s combinations: %w", combination.Name, err)
+		return c.manageFailure(ctx, combination, comboConditions.ManifestGenerationFailed, err)
+	}
+
+	log.Info(fmt.Sprintf("reconciliation of %s combination complete!", combination.Name))
+
+	// Return and update the combination's status
+	return c.manageSuccess(ctx, combination, generatedManifests)
+}
+
+// manageSuccess takes a combination CR and its evaluation to process a successful status update for it
+func (c *combinationController) manageSuccess(ctx context.Context, combination *v1alpha1.Combination, evaluation []string) (reconcile.Result, error) {
+	// Create the new status condition to transition to
+	combination.Status.Conditions = comboConditions.NewConditions(time.Now(), nil, SuccessCondition)
+
+	combination.Status.Evaluation = evaluation
+
+	// Update the status of the combination, requeue if this update fails
+	updateErr := c.Status().Update(ctx, combination)
+	if updateErr != nil {
+		c.log.Info("Error when updating success status, requeuing: " + updateErr.Error())
```

Comment:
should we log this as Error or Warn?

Reply:
Ah, good point. This looks like a place to use Error for sure.